### PR TITLE
Add tests for design.AttributeExpr.AllRequired()

### PR DESCRIPTION
**Pull request by tchssk**
#1423

### CODE REVIEW

File: design/attribute_test.go [hunk 1]
```diff
@@ -145,3 +145,50 @@ func TestAttributeExprValidate(t *testing.T) {
...
+			typ: &UserTypeExpr{
+				AttributeExpr: &AttributeExpr{
+					Validation: &ValidationExpr{
+						Required: []string{"foo", "bar"},
+					},
+				},
+			},
+			expected: []string{"foo", "bar"},
+		},
+		"not user type": {
```
Comment (raphael) [inline design/attribute_test.go]: I'm wondering if we need to test for this case? it works but having required attributes on a type that is not an object does not make a lot of sense.

Comment (tchssk) [reply design/attribute_test.go]: It makes sence. I remove the case and rename `user type` to `some required`.

Comment (tchssk) [reply design/attribute_test.go]: Fixed it. https://github.com/goadesign/goa/pull/1558/commits/581b2014316b876a5d61797b5bfc8bf0b7f8c7aa

Comment (raphael) [reply design/attribute_test.go]: Thank you!

**Pull request merged**